native/src/components/Header.tsx

```diff
@@ -213,18 +215,35 @@ const Header = ({
     if (currentRouteIsPoi && notFromDeepLink) {
       const poisRouteParams = route.params as RoutesParamsType[PoisRouteType]
       if (poisRouteParams.slug || poisRouteParams.multipoi !== undefined) {
-        return t('locations')
+        return { text: t('locations'), language: '' } // system language
       }
     }
 
     const previousRouteTitle = (previousParams as { title?: string } | undefined)?.title
-    return previousRouteTitle ?? t(previousRoute.name === 'categories' ? 'localInformation' : previousRoute.name)
+    let language
+    if (previousRoute.name === CATEGORIES_ROUTE && !(previousParams as { path?: string }).path) {
+      language = DEFAULT_LANGUAGE
+    } else if (previousRouteTitle || previousRoute.name === CATEGORIES_ROUTE) {
+      language = languageCode
+    } else {
+      language = '' // systen language
+    }
+
+    return {
+      text: previousRouteTitle ?? t(previousRoute.name === CATEGORIES_ROUTE ? 'localInformation' : previousRoute.name),
+      language,
+    }
```

Comment:
:wrench:
I think something like the following would reduce complexity:
```suggestion
    const previousRoutePath = (previousParams as { path?: string } | undefined)?.path

    if (previousRouteTitle) {
      return { text: previousRouteTitle, language: languageCode }
    }
    if (previousRoute.name === CATEGORIES_ROUTE) {
      return { text: t('localInformation'), language: previousRoutePath === undefined ? DEFAULT_LANGUAGE : languageCode }
    }
    return { text: previousRoute.name, language: undefined }
```

However, please check the if-clauses again, but basically I just took the old `text` return and compared with your if-clauses.

Reply:
Nice, thanks!